core/src/main/java/com/graphhopper/routing/template/ViaRoutingTemplate.java

```diff
@@ -132,8 +139,12 @@ public ViaRoutingTemplate(GHRequest ghRequest, GHResponse ghRsp, LocationIndex l
             // reset all direction enforcements in queryGraph to avoid influencing next path
             queryGraph.clearUnfavoredStatus();
 
-            if (algo.getVisitedNodes() >= algoOpts.getMaxVisitedNodes())
-                throw new IllegalArgumentException("No path found due to maximum nodes exceeded " + algoOpts.getMaxVisitedNodes());
+            if (algo.getVisitedNodes() >= algoOpts.getMaxVisitedNodes()) {
+                Map<String, Object> details = new HashMap<>();
+                details.put("max-visited-nodes", algoOpts.getMaxVisitedNodes());
+                throw new MaximumNodesExceededException("No path found due to maximum nodes exceeded", details);
```

Comment:
Why would we need this changed? What is the use case that a client can better parse the value?

Reply:
I think I overdid it with the map and details a bit :). But the different class, yes to make it easier for the client to recognize this exception and react accordingly.

Reply:
I actually think the number of maximum nodes is not really interesting for the client, so I removed it completely, WDYT?

Reply:
Yes, good idea. This is confusing to the client. Still maybe explain more what went wrong like "Path cannot be found as too many nodes ("junctions") already processed. This could happen if the locations are too far away" or something?

Reply:
> Still maybe explain more what went wrong like 

Good idea :+1: